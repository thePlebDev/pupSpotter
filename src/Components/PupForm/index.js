import React,{useState,useEffect,useRef} from 'react';
import {makeStyles} from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import MuiAlert from '@material-ui/lab/Alert';
import Button from '@material-ui/core/Button';
import styled from 'styled-components';

import NotificationSystem from '../Notification';



import useForm from '../../Hooks/useForm';
import {axiosPost} from '../../utils/AxiosFuncs'
import {Form} from '../../Assets/FormStylings';
import {signupValidation} from '../../utils/Validation';




const useStyles = makeStyles((theme) => ({
  root: {
    '& > *': {
      margin: theme.spacing(1),
      'font-size':'1.2em',
    },
    marginBottom:'20px',
  },
  alerts: {
    width: '100%',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
  },
  buttons: {
    width:'70%',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
    padding:'10px',
    border:'1px solid #05386B',
    marginBottom:'40px',
    marginLeft:'15%',


  },

}));

const Image = styled.img`
  display:none;
  @media screen and (min-width: 992px) {
  display:block;
}
`


function Alert(props) {
  return <MuiAlert elevation={6} variant="filled" {...props} />;
}

const PupForm =(props)=>{
  const classes = useStyles()
const {handleChange,handleClick,setOpen,show,status,handleSubmit,state,errors,setShow} = useForm(axiosPost,signupValidation)
const [loadingState,setLoadState] = useState(false)
const node = useRef();

  return(
    <div style={{height:'100%',display:'flex',alignItems:'center'}}>
      <Image src={require("./images/womanDog.png")} style={{width:'50%'}}  alt="dog in nature" />
      <Form onSubmit={(e)=>{handleSubmit(e)}} state={loadingState} enctype="multipart/form-data">
        <NotificationSystem status={status} show={show} setShow={setShow} />
      <div style={{margin:'0 auto',padding:'50px',borderRadius:'4px'}} >
      <div style={{color:'rgba(0, 0, 0, 0.5)',opacity:'0.8',textTransform:'uppercase',textAlign:'center',fontSize:'2.6em'}}>SPot a pup</div>
      <div style={{position:'relative',marginBottom:'30px'}}>
        <TextField inputProps={{ "data-testid": "content-input" }} className={classes.root} id="name-input" label="Dog Name" style={{width:'70%',marginLeft:'15%',padding:'5px'}}
        error={errors.name} data-testid="pupForm" value={state.name} onChange={(e)=>handleChange(e)} name='name'/>

      </div>
      <div style={{position:'relative',marginBottom:'50px'}}>
        <TextField className={classes.root} id="description-input" label="Description" style={{width:'70%',marginLeft:'15%'}}
        name="description" data-testid="pupForm" value={state.description}onChange={(e)=>handleChange(e)}/>
      </div>
        <div>

          <Button color={errors.loc?'secondary':'default'} className={classes.buttons} id="button" type="button" value={state.location}
          ref={node} onClick={(e)=>{handleClick()}} data-testid="pupForm" disableRipple={true} >Add Location</Button>
        </div>

        <div>
          <Button className={classes.buttons} data-testid="pupForm"  type="Submit" disableRipple={true}>Submit</Button>
        </div>
        </div>
      </Form>
    </div>
  )
}

export default PupForm
