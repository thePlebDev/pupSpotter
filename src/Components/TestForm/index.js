import React,{useState} from 'react';



const TestForm = ()=>{
  const [state,setState] = useState({test:''})
  const handleSubmit=(e)=>{
    e.preventDefault()
    console.log(state)
  }
  const handleChange =(e)=>{
    let {value,name} = e.target
    setState({...state,[name]:value})
    console.log(state)
  }

  return(
    <div>
      <form onSubmit={(e)=>{handleSubmit(e)}}>
        <label>
          <input id="name-input" value={state.test} name='test' onChange={(e)=>handleChange(e)}/>
        </label>
      </form>

    </div>
  )
}

export default TestForm
