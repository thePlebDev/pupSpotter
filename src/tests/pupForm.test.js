import React from 'react';
import PupForm from '../Components/PupForm';
import TestForm from '../Components/TestForm';
import renderer from 'react-test-renderer';
import { fireEvent } from "@testing-library/react"
import { shallow,mount } from 'enzyme';
import {BrowserRouter as Router} from 'react-router-dom';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({
  adapter: new EnzymeAdapter()
})
const simulateChangeOnInput = (wrapper,inputSelector,newValue,inputName)=>{
  const input = wrapper.find(inputSelector)
  input.simulate('change',{
    target:{name:inputName,value:newValue}
  })
  return wrapper.find(inputSelector)
}

describe('testing the pupForm',()=>{
  // it('testing the snapshot',()=>{
  //   const NotFoundComponent = renderer.create(<Router><PupForm/></Router>).toJSON()
  //   expect(NotFoundComponent).toMatchSnapshot();
  // })
  it('testing to make sure it renders',()=>{
    const component = shallow(<PupForm/>)
    const wrapper = component.find('[data-testid="pupForm"]')
    expect(wrapper.length).toBe(4)
  })
  it('tesitng the inputs',()=>{
    const wrapper = shallow(<PupForm/>)

    const updatedNameInput = simulateChangeOnInput(wrapper,'#name-input','Dave','name')
    const updatedDescInput = simulateChangeOnInput(wrapper,'#description-input','bueno boy','description')

    expect(updatedNameInput.props().value).toEqual('Dave')
    expect(updatedDescInput.props().value).toEqual('bueno boy')
  })
  it('testing the addLocation button',()=>{
    const wrapper = shallow(<PupForm/>);
    let addLocationBtn = wrapper.find('#button')
    addLocationBtn.simulate('click',{target:{
      name:'location',
      value:'turkey'
    }})
    addLocationBtn = wrapper.find('#button')
    expect(addLocationBtn.props().value).toEqual('eat')
  })
})
