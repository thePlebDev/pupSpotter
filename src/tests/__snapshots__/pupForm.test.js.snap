// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the pupForm testing the snapshot 1`] = `
<div>
  <form
    className="sc-AxhCb esycsr"
    onSubmit={[Function]}
  >
    <div
      style={
        Object {
          "border": "1px solid rgba(0, 0, 0, 0.87)",
          "borderRadius": "4px",
          "margin": "0 auto",
          "padding": "50px",
          "width": "60%",
        }
      }
    >
      <div
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.5)",
            "fontSize": "2.6em",
            "marginLeft": "40%",
            "opacity": "0.8",
            "textTransform": "uppercase",
          }
        }
      >
        SPot a pup
      </div>
      <div
        style={
          Object {
            "marginBottom": "30px",
            "position": "relative",
          }
        }
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
          data-testid="pupForm"
          style={
            Object {
              "marginLeft": "15%",
              "padding": "5px",
              "width": "70%",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="name-input"
            id="name-input-label"
          >
            Dog Name
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="name-input"
              name="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "marginBottom": "50px",
            "position": "relative",
          }
        }
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-root-1"
          data-testid="pupForm"
          style={
            Object {
              "marginLeft": "15%",
              "width": "70%",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="description-input"
            id="description-input-label"
          >
            Description
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="description-input"
              name="description"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-3"
          data-testid="pupForm"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
          value=""
        >
          <span
            className="MuiButton-label"
          >
            Add Location
          </span>
        </button>
      </div>
      <div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-buttons-3"
          data-testid="pupForm"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="Submit"
        >
          <span
            className="MuiButton-label"
          >
            Submit
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
